unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerHelper.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        /// <param name="bonesContainer">A SkinnedMeshRenderer that must contain the bones and bindposes that will be used by the combined avatar.</param>
+        /// <param name="renderersToCombine">A list of avatar parts to be combined</param>
+        /// <param name="materialAsset">A material asset that will serve as the base of the combine result. A new materialAsset will be created for each combined sub-mesh.</param>
+        /// <returns>true if succeeded, false if not</returns>
+        public bool Combine(SkinnedMeshRenderer bonesContainer, SkinnedMeshRenderer[] renderersToCombine, Material materialAsset)
+        {
+            Assert.IsTrue(bonesContainer != null, "bonesContainer should never be null!");
+            Assert.IsTrue(renderersToCombine != null, "renderersToCombine should never be null!");
+            Assert.IsTrue(materialAsset != null, "materialAsset should never be null!");
+
+            SkinnedMeshRenderer[] renderers = renderersToCombine;
+
+            // Sanitize renderers list
+            renderers = renderers.Where( (x) => x != null && x.enabled && x.sharedMesh != null ).ToArray();
+
+            if ( renderers.Length == 0 )
+                return false;
+
+            bool success = CombineInternal(
+                bonesContainer,
+                renderers,
+                materialAsset);
+
+            // Disable original renderers
+            for ( int i = 0; i < renderers.Length; i++ )
+            {
+                renderers[i].enabled = false;
+            }
+
+            return success;
+        }
+
+        private bool CombineInternal(SkinnedMeshRenderer bonesContainer, SkinnedMeshRenderer[] renderers, Material materialAsset)
+        {
+            Assert.IsTrue(bonesContainer != null, "bonesContainer should never be null!");
+            Assert.IsTrue(bonesContainer.sharedMesh != null, "bonesContainer should never be null!");
+            Assert.IsTrue(bonesContainer.sharedMesh.bindposes != null, "bonesContainer bindPoses should never be null!");
+            Assert.IsTrue(bonesContainer.bones != null, "bonesContainer bones should never be null!");
+            Assert.IsTrue(renderers != null, "renderers should never be null!");
+            Assert.IsTrue(materialAsset != null, "materialAsset should never be null!");
+
+            AvatarMeshCombiner.Output output = AvatarMeshCombiner.CombineSkinnedMeshes(
+                bonesContainer.sharedMesh.bindposes,
+                bonesContainer.bones,
+                renderers,
+                materialAsset);
+
+            if ( !output.isValid )
+            {
+                logger.LogError("AvatarMeshCombiner", "Combine failed!");
+                return false;
+            }
+
+            Transform rootBone = bonesContainer.rootBone;
+
+            if ( container == null )
+                this.container = new GameObject("CombinedAvatar");
```

Comment:
Yes, if the `AvatarMeshCombinerHelper` is disposed, the `gameObject` is destroyed. So if you call `Combine` after `Dispose` it will get a nullRef. The if prevents that.